app/models/stat.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Stat
+  def initialize(name, show_in_ui: false, expose_via_api: false, &block)
+    @name = name
+    @show_in_ui = show_in_ui
+    @expose_via_api = expose_via_api
+    @block = block
+  end
+
+  attr_reader :name, :expose_via_api, :show_in_ui
+
+  def calculate
```

Comment:
does it need to be public?

Reply:
Yes, it's called from a class method here:

https://github.com/discourse/discourse/blob/721b38c99c77769f39c5f036577710f692931b09/app/models/stat.rb#L44